### ISSUE TITLE
extract logic to `commons` lib and

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/android#10639]

**Comment by tobiasKaminsky** [comment]
This should be a task within Talk, as it was developed by Alvaro, so he is best to work further on it?
Actual usage of it, is then a task within Files.

**Comment by AlvaroBrey** [comment]
Yeah, this should be moved to the Talk repo, or to the common repo at least
